packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1514,3 +1561,48 @@ function legacyCreateInstances(cluster: DatabaseClusterNew, props: DatabaseClust
 function databaseInstanceType(instanceType: ec2.InstanceType) {
   return 'db.' + instanceType.toString();
 }
+
+/**
+ * Validate Performance Insights settings for legacy instanceProps with cluster level settings
+ */
+function legacyValidatePerformanceInsightsSettingsWithCluster(
```

Comment:
Can we create a single method using params? Idea
```ts
private validatePerformanceInsightsSettings(cluster: { 
    performanceInsightsEnabled: boolean, 
    performanceInsightRetention?: PerformanceInsightRetention, 
    performanceInsightEncryptionKey?: kms.IKey 
  }, instance: { 
    performanceInsightsEnabled: boolean, 
    performanceInsightRetention?: PerformanceInsightRetention, 
    performanceInsightEncryptionKey?: kms.IKey 
  }): void {
```

Reply:
done: https://github.com/aws/aws-cdk/pull/31385/commits/c0c937f6ef062c3c46df34087250f0a6070c377b

- `IConstruct` is needed for annotations, so `DatabaseClusterNew` was passed to cluster.
- Since `${instance.node.id}` is required in the error message, I added `nodeId` to the `instance`.
  - It is not needed in `instanceProps`, so it is left as an option.

```ts
 */
function validatePerformanceInsightsSettings(
  cluster: DatabaseClusterNew,
  instance: {
    nodeId?: string;
    performanceInsightsEnabled?: boolean;
    performanceInsightRetention?: PerformanceInsightRetention;
    performanceInsightEncryptionKey?: kms.IKey;
  },
): void {
```